src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -363,6 +364,14 @@ private function parseCallable($callable, $parameter, $id, $file)
                 return array($this->resolveServices($callable[0]), $callable[1]);
             }
 
+            if ('factory' === $parameter) {
```

Comment:
Should we blindly support this for configurators instead? IMHO it makes no sense for a service to configure itself.